webapp/channels/src/components/modal_controller/modal_controller.tsx

```diff
@@ -32,39 +32,39 @@ type Props = {
     };
 }
 
-export default class ModalController extends React.PureComponent<Props> {
-    public render(): React.ReactNode {
-        const {modals, ...props} = this.props;
-        const {modalState} = modals;
+const ModalController = ({
+    modals,
+    actions,
+}: Props) => {
+    const {modalState} = modals;
 
-        if (!modals) {
-            return null;
-        }
+    if (!modals) {
+        return null;
+    }
 
-        const modalOutput = [];
+    const modalOutput = [];
 
-        for (const modalId in modalState) {
-            if (modalState.hasOwnProperty(modalId)) {
-                const modal = modalState[modalId];
-                if (modal.open) {
-                    const modalComponent = React.createElement(modal.dialogType, Object.assign({}, modal.dialogProps, {
-                        onExited: () => {
-                            props.actions.closeModal(modalId);
+    for (const modalId in modalState) {
+        if (modalState.hasOwnProperty(modalId)) {
+            const modal = modalState[modalId];
+            if (modal.open) {
+                const modalComponent = React.createElement(modal.dialogType, Object.assign({}, modal.dialogProps, {
+                    onExited: () => {
+                        actions.closeModal(modalId);
 
-                            // Call any onExited prop provided by whoever opened the modal, if one was provided
-                            modal.dialogProps?.onExited?.();
-                        },
-                        onHide: props.actions.closeModal.bind(this, modalId),
-                        key: `${modalId}_modal`,
-                    }));
+                        // Call any onExited prop provided by whoever opened the modal, if one was provided
+                        modal.dialogProps?.onExited?.();
+                    },
+                    onHide: actions.closeModal.bind(this, modalId),
+                    key: `${modalId}_modal`,
+                }));
 
-                    modalOutput.push(modalComponent);
-                }
+                modalOutput.push(modalComponent);
             }
         }
-
-        return (
-            modalOutput
-        );
     }
-}
+
+    return modalOutput;
```

Comment:
@Syed-Ali-Abbas-Zaidi Regarding the type error that CI is hitting, the error message is rather cryptic, but the issue here is that we're returning an array of elements from a functional component. It can be fixed by wrapping this in a fragment though

```suggestion
    return <>{modalOutput}</>;
```